Remotion/Web/ClientScript/Scripts/Utilities.ts

```diff
@@ -377,44 +377,131 @@ class LayoutUtility
 {
   public static GetHeight(element: Element)
   {
-    return LayoutUtility.GetElementDimension(element, "height");
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "height");
+  }
+
+  public static GetInnerHeight(element: Element)
+  {
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "height", "padding");
   }
 
   public static GetWidth(element: Element)
   {
-    return LayoutUtility.GetElementDimension(element, "width");
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "width");
+  }
+
+  public static GetInnerWidth(element: Element)
+  {
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "width", "padding");
   }
 
-  private static GetElementDimension(element: Element, dimension: "width" | "height")
+  public static GetOuterHeight(element: HTMLElement)
+  {
+    return this.GetElementDimensionWithElementShow(element, "height", "border");
+  }
+
+  public static GetOuterWidth(element: HTMLElement)
+  {
+    return this.GetElementDimensionWithElementShow(element, "width", "border");
+  }
+
+  private static GetElementDimensionWithElementShow(element: Element, dimension: "width" | "height", extra?: "padding" | "border")
+  {
+    const htmlElement = element as HTMLElement;
+    const offsetValue = dimension === "width" ? htmlElement.offsetWidth : htmlElement.offsetHeight;
+    if (offsetValue !== 0)
+      return this.GetElementDimension(htmlElement, dimension, extra);
+
+    const previousPosition = htmlElement.style.position;
+    const previousVisibility = htmlElement.style.visibility;
+    const previousDisplay = htmlElement.style.display;
+
+    htmlElement.style.position = "absolute";
+    htmlElement.style.visibility = "hidden";
+    htmlElement.style.display = "block";
+
+    const result = this.GetElementDimension(htmlElement, dimension, extra);
+
+    htmlElement.style.position = previousPosition;
+    htmlElement.style.visibility = previousVisibility;
+    htmlElement.style.display = previousDisplay;
```

Comment:
Okay, you're forcing a re-draw here. I'm not sure that this doesn't result in flicker-sideeffects in the UI. There's definitely an effect to the DOM or updating the properties wouldn't be needed.

Reply:
That code was built to mirror the behavior of jQuery. Since we set and unset in the same JS call it should not redraw while in `this.GetElementDimension(...)`

Reply:
okay, great.